### DISCUSSION TITLE
Get the callback of on_search

### DISCUSSION THREAD

**Discussion by MiKoto-Railgun** [tomasklaen/uosc#1036] [Q&A] [answered]
I want to get the content in the input box in the menu. Now on_search can only return the content in the input box. I want to customize the callback so that the input content and some parameters I want are called back together

**Comment by MiKoto-Railgun** [comment]
Friends, please ask how to get the content of the input box in the uosc menu

`on_search = { "script-message-to", mp.get_script_name(), "update-data" }`

可以向`mp.register_script_message("update-data", function)`返回输入框里的内容
You can return the content in the input box to `mp.register_script_message("update-data", function)`

But it only returns the content of the input box. What if I want to customize the value it returns?

For example, `on_search = { "script-message-to", mp.get_script_name(), "update-data", "author" }` only returns the string "author", and the content in the input box cannot be called back. I want the return value to be similar to `on_search = { "script-message-to", mp.get_script_name(), "update-data", {"author", search_text} }` 

What should I do?

**Comment by tomasklaen** [comment, marked as answer]
I don't quite understand what you're trying to do, but as documented, if you're doing:

```lua
on_search = { "script-message-to", mp.get_script_name(), "update-data"}
```

Than query and menu id are appended as last parameters, so you get them like this:

```lua
mp.register_script_message("update-data", function(query, menu_id)
  -- here you'd probably do some searching/filtering of your data, and update like so
  local json = utils.format_json(new_menu_data)
  mp.commandv('script-message-to', 'uosc', 'update-menu', json)
end)
```

It's all documented in wiki: https://github.com/tomasklaen/uosc/wiki/Menu-API

**Reply by MiKoto-Railgun** [reply]
> 我不太明白你在做什么，但正如记录的那样，如果你正在这样做：
> 
> ```lua
> on_search = { "script-message-to", mp.get_script_name(), "update-data"}
> ```
> 
> Than query 和 menu id 作为最后一个参数附加，因此您可以像这样获取它们：
> 
> ```lua
> mp.register_script_message("update-data", function(query, menu_id)
>   -- here you'd probably do some searching/filtering of your data, and update like so
>   local json = utils.format_json(new_menu_data)
>   mp.commandv('script-message-to', 'uosc', 'update-menu', json)
> end)
> ```
> 
> 这一切都记录在 wiki 中：https://github.com/tomasklaen/uosc/wiki/Menu-API

I was stupid. I kept debugging and passed the parameters in array format, but it turned out that adding one more parameter would solve the problem. I was stupid.

PS：In on_search，the last parameter is the returned input search value.
